### PR TITLE
[redesign][ln] Send tab

### PR DESCRIPTION
**Pull request by bgptr**
Toward issue #3296

~~This diff updates only the header and the send form. I did not touch the history lists (Ongoing payments, Failed payments, Latest payments) since the design doesn't contain them.~~
update: @MariaPleshkova has added the list to the Figma designs, and this diff implements it.

Updated screenshots:
DeepinScreenshot_select-area_20210723100235
DeepinScreenshot_select-area_20210818143908
DeepinScreenshot_select-area_20210818143929
DeepinScreenshot_select-area_20210818143941

### CODE REVIEW

**Review by matheusd — COMMENTED** [review]
I feel you need to show more info about the decoded payment request. At the very least, you need to show the payment hash, description and the expiry.

Showing the extended details (ctlv expiry, fallback onchain address, etc) could be added in an expanded view.

It would probably be helpful to have the destination node use the copyable component as well.

**Comment by bgptr** [comment]
Thanks, @matheusd. I've added the suggested details and blocked attempting to pay for expired invoices.
Screenshots are updated. @MariaPleshkova, take a look at them if you'll have time, please.

File: app/components/views/LNPage/SendTab/DecodedPayRequest/helpers.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          id="ln.decodedPayRequestDetails.emptyFallbackAddr"
+          m="(empty fallback address)"
+        />
+      )
+    },
+    {
+      label: (
+        <T id="ln.decodedPayRequestDetails.paymentAddr" m="Payment Address" />
+      ),
+      value: decoded.paymentAddr,
```
Comment (matheusd) [inline app/components/views/LNPage/SendTab/DecodedPayRequest/helpers.js:23]: This should be converted to hex

File: app/components/views/LNPage/SendTab/DecodedPayRequest/DecodedPayRequest.jsx [hunk 2]
```diff
@@ -1,33 +1,109 @@
...
+
+    <div className={styles.paymentHashContainer}>
       <label>
-        <T id="ln.paymentsTab.destLabel" m="Destination" />
+        <T id="ln.paymentsTab.paymentHashLabel" m="Payment Hash" />:
       </label>
-      <div className={styles.destination}>
-        <TruncatedText text={decoded.destination} max={10} />
-      </div>
+      {decoded.paymentHash}
```
Comment (matheusd) [inline app/components/views/LNPage/SendTab/DecodedPayRequest/DecodedPayRequest.jsx:95]: This should be copyable

**Review by matheusd — COMMENTED** [review]
Looking good! Only two additional nits.

**Comment by bgptr** [comment]
@matheusd, thanks. I've pushed the changes.

**Comment by MariaPleshkova** [comment]
@bgptr generally looks good, but a few small tweaks from my side:
1. Changed "Expiry" title to the "Expiration Time" as we already use it this way.
2. Added a vertical stroke between "Amount-Destination" area and "Expiration Time".
3. "Description" and "Payment Hash" areas have to be visually similar to another info areas within this section (expiration time, destination, etc.). I mean that the titles should be located above their description text, not to the left from it.

image

**Comment by MariaPleshkova** [comment]
@bgptr and check the spacings within Lightning Payment Request Code, please. They should be 10px from all sides:
image

**Comment by MariaPleshkova** [comment]
And one more thing: let's show success and error states in a bit different way. 
You used orange and green icons(with an exclamation mark and a tick) on your screenshots, but we don't use them on the layouts in Figma for these fields. 
image
image

Instead we change the color of a field stroke to green and add a green description under it for success states, and change the stroke color+an entered text to orange and add an orange description under it as well for error states.
image
image

**Comment by bgptr** [comment]
@MariaPleshkova, thanks for the review. I've implemented your suggestions and updated the screenshots.

File: app/components/views/LNPage/SendTab/DecodedPayRequest/helpers.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { FormattedMessage as T } from "react-intl";
+export const getDecodedPayRequestDetails = (decoded) => {
+  const details = [
+    {
+      label: <T id="ln.decodedPayRequestDetails.ctlvExpiry" m="Ctlv Expiry" />,
```
Comment (matheusd) [inline app/components/views/LNPage/SendTab/DecodedPayRequest/helpers.js:5]: CLTV instead of ctlv

**Review by matheusd — APPROVED** [review]
Final typo, but otherwise LGTM.

File: app/components/views/LNPage/SendTab/DecodedPayRequest/helpers.js (hunk 3 above)
Comment (bgptr) [reply app/components/views/LNPage/SendTab/DecodedPayRequest/helpers.js]: Thanks, @matheusd. Fixed.

**Review by matheusd — APPROVED** [review]

**Pull request merged**